### PR TITLE
Standardize Logger Implementation with Loguru and Add Comprehensive Unit Tests

### PR DESCRIPTION
**Pull request by markurtz**
## Summary
Standardizes the logger implementation around the loguru library and latest standards for NM. It enhances logging flexibility and configurability and introduces comprehensive unit tests to ensure reliability.

## Details
- Introduced a new logger.py module that utilizes the loguru library.
- The logger configuration supports console and file logging, with settings customizable via environment variables or direct function calls.
- Environment variables include options for disabling logging, clearing existing loggers, setting log levels for console and file logging, and specifying log file paths.
- Updated __init__.py to expose the logger configuration functions and logger instance.
- Added detailed docstrings and usage examples in logger.py.
- Implemented comprehensive unit tests in test_logger.py to cover various configurations and scenarios, ensuring robust functionality.

## Test Plan
Automation Testing:
- Ran pytest to execute the newly added unit tests in test_logger.py.
- Verified that default, environment variable overrides, and explicit configurations are correctly applied and logged.
Manual Testing:
- Manually tested logging outputs to console and file to ensure proper formatting and log levels are respected.

### CODE REVIEW

File: tests/unit/test_logger.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+from guidellm import LoggerConfig, configure_logger, logger
+
+
+@pytest.fixture(autouse=True)
+def reset_logger():
+    # Ensure logger is reset before each test
+    logger.remove()
+    yield
+    logger.remove()
```
Comment (parfeniukink) [inline tests/unit/test_logger.py:11]: I believe that we could keep in in a `tests/conftest.py` so this fixture is available for the whole project. What do you think?

Comment (parfeniukink) [reply tests/unit/test_logger.py:11]: Another thing that we could bring up is using of the `pytest_configure` reserved hook that could include the logger configuration.

```py
# /tests/conftest.py
def pytest_configure() -> None:
    logger.disable("guidellm")
```

**Comment by parfeniukink** [comment]
@markurtz
Side topic: "What about using the `pydantic` for the settings and environment variables management in thr project? It is still to be a dependency for the `openai` package btw"

**Comment by markurtz** [comment]
> @markurtz Side topic: "What about using the `pydantic` for the settings and environment variables management in thr project? It is still to be a dependency for the `openai` package btw"

Good idea, I'm going to push this off, though, to another PR once the pydantic one lands and we can refactor around that

File: tests/unit/test_logger.py (hunk 1 above)
Comment (markurtz) [reply tests/unit/test_logger.py:11]: Good idea, I'm going to push this off for now to another PR to combine with the pydantic env management

**Pull request merged**